### DISCUSSION TITLE
use pyaedt to export SummaryReport

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#4687] [Q&A] [answered]
Hi all, I used pyaedt to export several SummaryReport data such as "Temperature" and "HeatFlowRate" during post-processing and it helped me save a lot of time! 
But I have a small question.The question is that when exporting SummaryReaport using AEDT Icepak GUI, the "Side" option has three choices, namely "Default", "Ajacent" and "Combined", usually the results of these three options will be different.
There is also an option to "Include all surface meshes". As you can see in the picture below.
![image](https://github.com/ansys/pyaedt/assets/163245306/303dde64-9504-4b19-9d6f-6e7261235486)
However, when I use pyaedt to export SummaryReport, I use the following code, this code successfully exported the SummaryReport, but the column "Side" is defaulted to "Default" and the column "Mesh" is defaulted to "Reduced".
```
ipk.export_summary(output_dir=file_path,type="Boundary",geometryType="Surface",quantity="HeatFlowRate")
```
![image](https://github.com/ansys/pyaedt/assets/163245306/5c774de9-0413-44b1-930b-79014a8ce787)
My requirement is to set the "Side" column to "Combined" and the "Mesh" column to "All". I would like to know how to modify my code for this.  
Looking forward to your reply, thank you!

**Comment by Samuelopez-ansys** [comment]
@lorenzovecchietti

**Comment by lorenzovecchietti** [comment, marked as answer]
Hi @aknsj, I introduced a new class that I think can help you in this case.

```
temperature_fs = ipk.post.create_field_summary()
temperature_fs.add_calculation(
        entity="Object",
        geometry="Surface",
        geometry_name="FaceList1",
        quantity="Temperature",
        side="Default",
        mesh="All",
    )
temperature_fs = temperature_fs.get_field_summary_data(pandas_output=True)
```

With this you'll have a pandas dataframe that allows you to do whatever post-processing in python directly using the power of pandas.

If you are interested instead only to the csv export, after you create the field summary object, you can execute:
`temperature_fs.export_csv(output_file="test.csv")`

**Reply by aknsj** [reply]
Thanks for your reply, I tried the script you provided, but the following Error occurs: 'PostProcessor' object has no attribute 'create_field_summary'.Looking forward to your reply!

**Reply by lorenzovecchietti** [reply]
Please check if you are using the latest version of pyaedt.

**Reply by aknsj** [reply]
Hi, I checked the version of pyaedt I am currently using and it is 0.7.3. Then I used the following command to detect that the latest version of pyaedt is 0.9.2 and pyaedt was upgraded to this version.
```
pip list --outdated
```
But the following problem still exists.
```
'PostProcessor' object has no attribute 'create_field_summary'
```
Looking forward to your reply, thanks!

**Reply by Samuelopez-ansys** [reply]
Hi @aknsj ,

Please do at the beggining of your code:

import pyaedt

print(pyaedt.__version__)

Because very likely you are using an old PyAEDT version. In the code, we can see it, check the unit test for instance:
https://github.com/ansys/pyaedt/blob/main/_unittest_solvers/test_00_analyze.py#L196

Regards

**Reply by aknsj** [reply]
Hi, thanks for the reply. I got the version of pyaedt using the following script.
![image](https://github.com/ansys/pyaedt/assets/163245306/0c8d65e8-a477-4477-bf04-c5bb8fa1e72d)
Then I successfully set up the side and mesh of the summary report and exported the dataframe as a .csv file using the following script.
```
temperature_fs = ipk.post.create_field_summary()
temperature_fs.add_calculation(
        entity="Object",
        geometry="Surface",
        geometry_name="FaceList",
        quantity="Temperature",
        side="Combined",
        mesh="All",
    )
csv_name = 'summary_report.csv'
temperature_fs.export_csv(output_file=os.path.join(base_path,csv_name))
```
Thank you guys, your answers really helped me!
